core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2MetadataAsyncClient.java

```diff
@@ -52,41 +49,35 @@
 import software.amazon.awssdk.utils.Logger;
+import software.amazon.awssdk.utils.ThreadFactoryBuilder;
+import software.amazon.awssdk.utils.Validate;
 
 @SdkInternalApi
 @Immutable
 @ThreadSafe
-public final class DefaultEc2MetadataAsyncClient implements Ec2MetadataAsyncClient {
+public final class DefaultEc2MetadataAsyncClient extends BaseEc2MetadataClient implements Ec2MetadataAsyncClient {
 
     private static final Logger log = Logger.loggerFor(DefaultEc2MetadataClient.class);
     private static final int DEFAULT_RETRY_THREAD_POOL_SIZE = 3;
 
-    private final Ec2MetadataRetryPolicy retryPolicy;
     private final SdkAsyncHttpClient httpClient;
-    private final EndpointMode endpointMode;
-    private final URI endpoint;
-    private final RequestMarshaller requestMarshaller;
-    private final Duration tokenTtl;
     private final ScheduledExecutorService asyncRetryScheduler;
+    private final boolean isHttpClientManaged;
+    private final boolean isRetryExecutorManaged;
 
     private DefaultEc2MetadataAsyncClient(Ec2MetadataAsyncBuilder builder) {
-        this.retryPolicy = builder.retryPolicy != null ? builder.retryPolicy
-                                                       : Ec2MetadataRetryPolicy.builder().build();
-        this.endpointMode = builder.endpointMode != null ? builder.endpointMode
-                                                         : DEFAULT_ENDPOINT_PROVIDER.resolveEndpointMode();
-        this.endpoint = builder.endpoint != null ? builder.endpoint
-                                                 : URI.create(DEFAULT_ENDPOINT_PROVIDER.resolveEndpoint(this.endpointMode));
-        this.requestMarshaller = new RequestMarshaller(this.endpoint);
-        this.tokenTtl = builder.tokenTtl != null ? builder.tokenTtl
-                                                 : Duration.ofSeconds(21_600);
-        this.asyncRetryScheduler = builder.scheduledExecutorService != null
-                                   ? builder.scheduledExecutorService
-                                   : Executors.newScheduledThreadPool(DEFAULT_RETRY_THREAD_POOL_SIZE);
-        this.httpClient = builder.httpClient != null
-                          ? builder.httpClient
-                          : new DefaultSdkAsyncHttpClientBuilder().buildWithDefaults(AttributeMap.empty());
+        super(builder);
+        this.httpClient = Validate.getOrDefault(builder.httpClient,
+            () -> new DefaultSdkAsyncHttpClientBuilder().buildWithDefaults(AttributeMap.empty()));
+        this.asyncRetryScheduler = Validate.getOrDefault(builder.scheduledExecutorService,
+            () -> {
+                ThreadFactory threadFactory = new ThreadFactoryBuilder().threadNamePrefix("IMDS-ScheduledExecutor").build();
+                return Executors.newScheduledThreadPool(DEFAULT_RETRY_THREAD_POOL_SIZE, threadFactory);
+            });
+        this.isHttpClientManaged = builder.httpClient == null;
```

Comment:
Does `managed` mean managed by us? I first thought it was managed by the user but it was the opposite. Can we change it to `isInternal...` or `isDefault...` instead?